File: Assets/QuestScript.js
#pragma strict
import System.Collections.Generic; 
class QuestScript {

    protected var quest : Quest;
    protected var currentStage : int;
    protected var objectives : List.<QuestObjective>;

    function QuestScript(quest:Quest) {
        this.quest = quest;
        this.currentStage = quest.Stage;
        this.objectives = quest.QuestObjectives;

    }
    function Start () {

    }

    function Update () {

        if(currentStage != quest.Stage) {
            currentStage = quest.Stage;
            OnStageChange(quest.Stage);
        }

    }
    function OnStageChange(stage:int) {

    }

}

File: Assets/Standard Assets/Quest/QuestScript.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class QuestScript {

    protected Quest quest;
    protected int currentStage;
    private static Dictionary<string, object> questVars; //structure for representing name of quest variable and value of it.  We don't know how many or what types they're going to hold.

    public QuestScript(Quest quest) {
        this.quest = quest;
        currentStage = quest.Stage;
        questVars = new Dictionary<string, object>();
    }
	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
	
	}
   protected void AddQuestVariable(string varName, object value) {
        /*
        Add a quest variable along with a value
        */
        questVars.Add(varName, value);
    } 

    protected void UpdateQuestVariable (string varName, object value) {

        questVars[varName] = value;

    }
}
